Pm 29 create shelter shelter id part 2 (edit and delete)

dotte-osu opened the pull request:
Added "Edit" and "Delete" shelter functions.

Edit part:
![image](https://user-images.githubusercontent.com/71732016/155250927-fbe5c7e1-19e6-4512-b2fb-1242cef74a36.png)

I didn't add password change because that is more like shelter "account" data change rather than shelter "information" data. Now that I think about it.... I am not sure if we should use this "login email address" as their shelter "contact email address". (sorry, I am not sure how I should describe this...let me know if this is confusing.) Maybe we should have an optional "contact email address" column in the database?...but that also seems confusing. Maybe it's okay as it is.

Delete part:
If a shelter wants to delete their shelter data, the page will give them a warning pop up. 
![image](https://user-images.githubusercontent.com/71732016/155251306-cd19707d-6e3a-4509-8ebf-d7786ecdd5f6.png)
Once the shelter click `OK`, the page will take them to the home page and the shelter is automatically logged out.

dotte-osu committed:
added edit shelter function

dotte-osu committed:
fixed small issues

dotte-osu committed:
removed unused code

dotte-osu committed:
added delete shelter function

dotte-osu committed:
removed unused code

dotte-osu commented on src/components/CreateAccountForm/CreateAccountForm.js:
```diff
@@ -4,7 +4,7 @@ import AuthContext from '../../context/AuthContext';
 import SheltersService from '../../services/sheltersService';
 import PasswordChecklist from 'react-password-checklist'
-import USStates from './USStates.json'
+import USStates from '../../config/USStates.json'
```
moved the json file to config folder

xHoudek replied:
Moving it is a good idea! Do you think it should go into utils instead? It isn't really something that is configurable

dotte-osu replied:
ah! yeah utils is better! I will move it.

dotte-osu commented on src/routes/SheltersPege/SheltersPege.js:
```diff
@@ -21,35 +27,52 @@ export default class SheltersPege extends Component {
     };
     this.inputChanged = this.inputChanged.bind(this);
+    this.onClickEdit = this.onClickEdit.bind(this);
+    this.onClickUpdate = this.onClickUpdate.bind(this);
```
This actually update the database

dotte-osu commented on src/routes/SheltersPege/SheltersPege.js:
```diff
@@ -21,35 +27,52 @@ export default class SheltersPege extends Component {
+      error: null
     };
     this.inputChanged = this.inputChanged.bind(this);
+    this.onClickEdit = this.onClickEdit.bind(this);
```
This switches the "view" mode and "edit" mode

xHoudek replied:
I think it would be more clear if we changed this name. Maybe something like `handleEditButtonClick`? Adding code comments above the two functions could also be good for clarification

dotte-osu replied:
fixed!

dotte-osu committed:
Merge branch 'main' into PM-29-create-shelter-shelter-id-part-2

xHoudek commented on src/routes/SheltersPege/SheltersPege.js:
```diff
@@ -1,15 +1,21 @@
+import USStates from '../../config/USStates.json'
 var log = logUtils.getLogger()
 
 export default class SheltersPege extends Component {
```
Not part of this PR, but something I noticed: This should be `SheltersPage` instead of `SheltersPege`. Same with the filename

dotte-osu replied:
fixed

xHoudek commented on src/routes/SheltersPege/SheltersPege.js:
```diff
@@ -77,34 +100,210 @@ export default class SheltersPege extends Component {
     return filters;
   }
 
+  updateState(shelter) {
```
`this.setState` can take an object with multiple attributes, so you probably only need one call to `this.setState`

```
this.setState({
  shelterName: shelter[0].ShelterName,
  address: shelter[0].Address.substring(0, shelter[0].Address.indexOf(',')),
  city: shelter[0].Address.substring(shelter[0].Address.indexOf(',') + 2, shelter[0].Address.lastIndexOf(',')),
  state: shelter[0].Address.substring(shelter[0].Address.lastIndexOf(',') + 2, shelter[0].Address.lastIndexOf(',') + 4),
  zip: parseInt(shelter[0].Address.substring(shelter[0].Address.length - 5)),
  email: shelter[0].EmailAddress,
  phoneNumber: parseInt(shelter[0].PhoneNumber),
  website: shelter[0].Website
});
```

dotte-osu replied:
ah! pretty! thank you!

xHoudek commented on src/routes/SheltersPege/SheltersPege.js:
```diff
@@ -21,35 +27,52 @@ export default class SheltersPege extends Component {
+      })
+      .catch(error => {
+        log.error(error)
+        this.setState({ error: "No shelter data avairable" });
```
Small typo: should be "available"

xHoudek commented on src/routes/SheltersPege/SheltersPege.js:
```diff
@@ -77,34 +100,210 @@ export default class SheltersPege extends Component {
+      "shelterName": shelterName,
+      "address": address + ", " + city + ", " + state + " " + zip,
+      "emailAddress": email,
+      "phoneNumber": parseInt(phoneNumber),
```
Is `parseInt()` necessary? I thought the DB was now storing this value as a string

dotte-osu replied:
good catch.

xHoudek replied:
Looks like this wasn't updated

xHoudek commented on src/routes/SheltersPege/SheltersPege.js:
```diff
@@ -77,34 +100,210 @@ export default class SheltersPege extends Component {
+  }
+
+  renderUSStates() {
+    let stateOptions = [<options></options>];
```
`<options></options>` can be removed. I added that in there before the states were moved to a JSON, where there is an empty string built in. Also the `<options>` tag doesn't exist, which is totally my bad lol

dotte-osu replied:
fixed

xHoudek commented on src/services/sheltersService.js:
```diff
@@ -29,6 +30,35 @@ const SheltersService = {
         : res.json());
+  },
+  updateShelter(shelter) {
+    console.log("postShelter")
```
console logs can be removed

dotte-osu replied:
removed

xHoudek commented on src/services/sheltersService.js:
```diff
@@ -29,6 +30,35 @@ const SheltersService = {
+        'content-type': 'application/json',
+        'authorization': AuthService.getToken()
+      },
+      body: JSON.stringify({ ...shelter })
```
This is a question for my own edification. Why use `{ ...shelter}` instead of `shelter`? I still have yet to fully grasp the usage of `...` in JS

dotte-osu replied:
In this case, ... formats an array to json nicely (I believe but I might be wrong!), but I am simply passing a json object so I removed it!

xHoudek commented on src/routes/SheltersPege/SheltersPege.js:
```diff
@@ -77,34 +100,210 @@ export default class SheltersPege extends Component {
+          <div className="row">
+            <div className="col-sm-12 text-center">
+              <button className="btn btn-primary btn-md m-2" onClick={this.onClickUpdate}>Update</button>
+              <button className="btn btn-success  btn-md m-2" onClick={this.onClickEdit}>Cancel</button>
```
I think this should be changed to `className="btn btn-outline-primary"`

dotte-osu replied:
much prettier! thank you!

xHoudek reviewed (changes requested):
I'm struggling to get this to function properly when I test. I log in as a shelter and am able to access the edit page, but when I change something and click "Update", nothing happens. It looks like all of the information stays the same and the information doesn't actually get updated. Not sure if that is something on my end or what.

Also, every time I type a new character in on of the edit inputs, a new GET request is made. I think we should change it so that GET request isn't being made every time.

Overall though, I really like they layout and the way the edit section fits into the shelter page! Great design. Just some kinks to iron out.

Delete functionality seems to work as expected. Pets associated with the shelter are also removed, which is great!

xHoudek commented:
After some discussion, it looks like the unable to update issue may be on my end (most likely is). @asching7108 could you test this, along with [the backend PR](https://github.com/asching7108/purrfect-match-api/pull/30), to see if you're seeing the same issue I am?

dotte-osu committed:
fixed option issue

dotte-osu committed:
fixed multiple GET call issue

dotte-osu committed:
moved json file

dotte-osu committed:
addressed code comments

dotte-osu commented:
Addressed the code comments. Let me know if the code doesn't work.

xHoudek commented on src/routes/SheltersPege/SheltersPege.js:
```diff
@@ -1,14 +1,20 @@
 var log = logUtils.getLogger()
 
-export default class SheltersPege extends Component {
+export default class SheltersPage extends Component {
```
I think the folder and Files need to be renamed to say "SheltersPage" too. Same with the .css file

xHoudek reviewed (changes requested):
Looks like there are a few more minor typos and things to change. I am still unable to update anything, just waiting for Esther to test

dotte-osu committed:
fixed SheltersPage names

dotte-osu commented:
Sorry, fixed all the "SheltersPage" issues. I also re-clone the repo and tested in Chrome and Firefox. The update function works fine. If `SheltersService.updateShelter` is called, the issue must be in the backend. Most likely update sql query because I updated the phone number data type to varchar(10) and the error message should show up in the backend console.

dotte-osu committed:
removed parseInt

dotte-osu committed:
added max

dotte-osu committed:
Merge branch 'main' into PM-29-create-shelter-shelter-id-part-2

xHoudek commented on src/routes/SheltersPage/SheltersPage.js:
```diff
@@ -0,0 +1,380 @@
+      })
+      .catch(error => {
+        log.error(error)
+        this.setState({ error: "No shelter data avairable" });
```
This typo still needs to be fixed I think

dotte-osu committed:
Merge branch 'main' into PM-29-create-shelter-shelter-id-part-2

dotte-osu committed:
fixed patch issue with FireFox

dotte-osu committed:
fixed typo

dotte-osu committed:
added required tags to the inputs

xHoudek reviewed (approved):
Great job on this! Glad we were able to iron out the kinks
